recipes/r-bpcells/meta.yaml

```diff
@@ -0,0 +1,164 @@
+{% set version = '0.3.0' %}
+{% set github = 'https://github.com/bnprks/BPCells' %}
+{% set posix = 'm2-' if win else '' %}
+
+package:
+  name: r-bpcells
+  version: {{ version|replace('-', '_') }}
+
+source:
+  url: {{ github }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 231e91a3d60d2bc400324c5b60fb4644fda16261048d6b1a49bfac4aa269cd1f
+  patches:
+    # track: https://github.com/bnprks/BPCells/issues/228
+    - patches/0001-fix-libhwy-config.patch
+    # from: https://github.com/bnprks/BPCells/pull/231
+    - patches/0002-granges-conversion.patch
+
+build:
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  run_exports:
+    - {{ pin_subpackage('r-bpcells', max_pin='x.x') }}
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - cross-r-base {{ r_base }}  # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - r-base
+    - r-dplyr >=1.0.0
+    - r-ggplot2 >=3.4.0
+    - r-ggrepel
+    - r-hexbin
+    - r-lifecycle
+    - r-magrittr
+    - r-matrix
+    - r-patchwork
+    - r-rcolorbrewer
+    - r-rcpp
+    - r-rcppeigen
+    - r-readr
+    - r-rlang
+    - r-scales
+    - r-scattermore
+    - r-stringr
+    - r-tibble
+    - r-tidyr
+    - r-vctrs
+    - hdf5
+    - libhwy
+    - zlib
+  run:
+    - r-base
+    - r-dplyr >=1.0.0
+    - r-ggplot2 >=3.4.0
+    - r-ggrepel
+    - r-hexbin
+    - r-lifecycle
+    - r-magrittr
+    - r-matrix
+    - r-patchwork
+    - r-rcolorbrewer
+    - r-rcpp
+    - r-readr
+    - r-rlang
+    - r-scales
+    - r-scattermore
+    - r-stringr
+    - r-tibble
+    - r-tidyr
+    - r-vctrs
+
+test:
+  requires:
+    - r-testthat
+    - bioconductor-genomicranges
+    - bioconductor-iranges
+    - r-igraph
+    - r-matrixstats
+    - r-rspectra
+    - r-withr
+    - macs3
+  commands:
+    - $R -e "library('BPCells')"
+    ## NB: this tightly-couples to `macs2` coming from `test.requires` and avoids running against the container
+    - if command -v macs3; then $R -e "testthat::test_package('BPCells', stop_on_failure=TRUE)"; fi
```

Comment:
I'm confused why the `if command -v macs3` is necessary. The `test.requires` installs `macs3`, so how could it be unavailable? And where does `macs2` come in? I thought the container was very minimal, so I didn't think it would have either macs2 or macs3 pre-installed.

Reply:
The `test.requires` dependencies only get installed in the regular `conda-build` test, but the container test cannot install them and will fail if one attempts to run a test that requires them. This is just a proxy for knowing whether we are in the container. See https://github.com/bioconda/bioconda-recipes/issues/52189.

The "`macs2`" is just that I switched here to `macs3` but didn't update the comment. I can update that to avoid confusion.

Reply:
Thanks - I think I'll merge for now and just remember to touch it up later. The patches here are now upstream, so there will be other stuff to touch up when they release next.